Final project od Sprint_4

MikhailZakharov1 opened the pull request:
Sprint_4

Commit:
Final project od Sprint_4

ZeTscor commented on src/main/java/ru/yandex/praktikum/HomePage.java:
```diff
@@ -0,0 +1,114 @@
+    //локатор верхней кнопки для заказа "Заказать"
+    private By higherOrderButton = By.xpath(".//button[(@class ='Button_Button__ra12g' and text()='Заказать')]");
+    //локатор нижней кнопки для заказа "Заказать"
+    private By lowerOrderButton = By.className("Button_Middle__1CSJM");
```
Локатор не верен, так как он динамический и при открытие браузера на весь экран меняется, нужно либо якорный индекс или отталкивается от родительского элемента.

MikhailZakharov1 replied:
Сменил локатор нижней кнопки на xpath добавив локатор
.//div[@class ='Home_FinishButton__1_cWm']//button

ZeTscor commented on src/test/java/ru/yandex/praktikum/scooterTests/OrderStatusTest.java:
```diff
@@ -0,0 +1,77 @@
+                .clickOkButton();
+        //создаю объект страницы Заказ оформлен
+        OrderIsWork objOrderIsProcessed = new OrderIsWork(driver);
+        //проверка, что поле "Заказ оформлен". В приложении есть баг, который не даёт оформить заказ. Он воспроизводится только в Chrome
```
Хорошо, что есть комментарии.

ZeTscor commented on src/test/java/ru/yandex/praktikum/scooterTests/ConnectionChrome.java:
```diff
@@ -0,0 +1,28 @@
+import java.util.concurrent.TimeUnit;
+
+//базовый класс для подключения драйвера Chrome
+public class ConnectionChrome {
```
Лучше не связывать названия с браузером.

MikhailZakharov1 replied:
Исправил на Connection

ZeTscor commented on src/test/java/ru/yandex/praktikum/scooterTests/ConnectionChrome.java:
```diff
@@ -0,0 +1,28 @@
+
+    //подключение драйвера браузера и установление ожидания в 5 секунд
+    @Before
+    public void open() {
```
Тут лучше реализовать фабрику для того что бы сразу была возможность запустить firefox.

MikhailZakharov1 replied:
Про "Фабрику" не было и слова в учебном материале, а так же в задании

ZeTscor commented on src/test/java/ru/yandex/praktikum/scooterTests/ConnectionChrome.java:
```diff
@@ -0,0 +1,28 @@
+    //подключение драйвера браузера и установление ожидания в 5 секунд
+    @Before
+    public void open() {
+        WebDriverManager.chromedriver().setup();
```
Отлично, что использован webdrivermanager, единственный момент его лучше обновить в pom.

MikhailZakharov1 replied:
Если правильно понял обновил в pom конфиг Selenium на 4.27.0, в итоге пришлось изменить Тайминг на Duration.ofSeconds(5) (которого не было в учебном материале)

ZeTscor reviewed (commented):
Работа хорошая, но надо поменять локатор для кнопки внизу, и изменить названия файла конфигурации драйвера.

Commit:
1.Change locator lowerOrderButton
2.Update pom configurationSelenium-java
3.Refactor WebDriverWait to Duration.ofSeconds()
